share/smack/include/pthread.h

```diff
@@ -0,0 +1,71 @@
+//
```

Comment:
We have this line:
https://github.com/smackers/smack/blob/pthreadsFeature/bin/smack.py#L252
Using `-I` appends the folder to the beginning of the compiler's search list, and so Monty's `pthread.h` should be picked up before the system's `pthread.h`. This part makes sense to me and I don't see why it would not work. We should test it be absolutely sure though.
Now, I am confused about the `pthreadtypes.h` part. Do user files ever include `pthreadtypes.h` or is it included just from `pthread.h'? It seems that Monty hinted that: (a) user files do include it and on top of that (b) they include it using implementation dependent paths/names. Is that right? Monty, could you maybe point out some example?